### PR TITLE
Feat/add points to websocket

### PR DESCRIPTION
**Pull request by mokshgrappus007**

### CODE REVIEW

File: api/packages/player_repository/lib/models/drawing_points.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  bool get isNull => points == null;
+
+  Map<String, dynamic> toJson() {
+    return {
+      'points': points?.toJson(),
+      'paint': paint?.toJson(),
+    };
+  }
+
+  factory DrawingPointsWrappper.fromJson(Map<String, dynamic> json) {
```
Comment (noorbhatia10) [inline api/packages/player_repository/lib/models/drawing_points.dart]: Wrapper spelling

File: packages/game_repository/.github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!--
+  Thanks for contributing!
```
Comment (noorbhatia10) [inline packages/game_repository/.github/PULL_REQUEST_TEMPLATE.md]: remove .github from packages

File: api/lib/session/bloc/session_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    on<OnPlayerAdded>(_onPlayerAdded);
+    on<OnPointsAdded>(_onAddPoints);
+  }
+
+  void _onPlayerAdded(OnPlayerAdded event, Emitter<SessionState> emit) {
+    var players = <Player>[...state.players, event.player];
+    emit(
+      state.copyWith(
+        players: players,
+        playerId: event.player.userId,
```
Comment (noorbhatia10) [inline api/lib/session/bloc/session_bloc.dart]: playerId -> currentPlayerId

File: packages/game_repository/.github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mokshgrappus007) [reply packages/game_repository/.github/PULL_REQUEST_TEMPLATE.md]: okh :+1:

File: api/lib/session/bloc/session_event.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  List<Object?> get props => [player];
+}
+
+class OnPointsAdded extends SessionEvent {
+  const OnPointsAdded(this.points);
+
+  final DrawingPointsWrappper points;
+
+  @override
+  List<Object?> get props => [points];
```
Comment (noorbhatia10) [inline api/lib/session/bloc/session_event.dart:22]: Add an event for OnPlayerDisconnected

File: api/routes/ws.dart [hunk 5]
```diff
@@ -1,29 +1,30 @@
...
-            final event = AddMousePointerEvent.fromMap(map);
-            offsetCubit.addOffset(Point(event.mouseX, event.mouseY));
-          default:
-        }
-      }
+    channel.stream.listen((data) {
+      final jsonData = jsonDecode(data.toString());
+      final receivedPoints =
+          DrawingPointsWrappper.fromJson(jsonData as Map<String, dynamic>);
+      sessionBloc.add(OnPointsAdded(receivedPoints));
```
Comment (noorbhatia10) [inline api/routes/ws.dart]: Handle OnPlayerDisconnected when stream is closed in onDone() method

File: api/lib/session/bloc/session_bloc.dart [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import 'package:player_repository/player_repository.dart';
+
+part 'session_event.dart';
+
+part 'session_state.dart';
+
+class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
+  SessionBloc() : super(const SessionState()) {
+    on<OnPlayerAdded>(_onPlayerAdded);
+    on<OnPointsAdded>(_onAddPoints);
```
Comment (noorbhatia10) [inline api/lib/session/bloc/session_bloc.dart:14]: OnPlayerDisconnected will remove the player from players list

Comment (mokshgrappus007) [reply api/lib/session/bloc/session_bloc.dart:14]: @noorbhatia  done with issues :heavy_check_mark:

File: api/routes/ws.dart [hunk 7]
```diff
@@ -20,12 +20,18 @@ Future<Response> onRequest(RequestContext context) async {
...
+    channel.stream.listen(
+      (data) {
+        final jsonData = jsonDecode(data.toString());
+        final receivedPoints =
+            DrawingPointsWrapper.fromJson(jsonData as Map<String, dynamic>);
+        sessionBloc.add(OnPointsAdded(receivedPoints));
+      },
+      onDone: () {
+        final currentUserId = sessionBloc.state.currentPlayerId;
+        sessionBloc.add(OnPlayerDisconnect(currentUserId));
```
Comment (noorbhatia10) [inline api/routes/ws.dart]: also unsubscribe the broadcast bloc -> sessionBloc..add(OnPlayerDisconnected)..unsubscribe()

Comment (mokshgrappus007) [reply api/routes/ws.dart]: check now @noorbhatia10

**Review by sarthak-grappus — APPROVED** [review]

**Pull request merged**